olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,149 @@
+            ),
+            "static_groups": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use static groups for quantization. Default value is False.",
+            ),
+            "true_sequential": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use true sequential for quantization. Default value is False.",
+            ),
+            "desc_act": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is True.",
+            ),
+            "gpu": PassConfigParam(
+                type_=bool,
+                default_value=True,
+                description="Use GPU for quantization. Default value is True.",
+            ),
+            "export_optimization": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Export optimization level. Default value is None.",
+            ),
+        }
+        config.update(get_external_data_config())
+        return config
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from optimum.exporters.onnx import onnx_export_from_model
+        from transformers import AutoModelForCausalLM, AutoTokenizer, GPTQConfig
+
+        tokenizer = AutoTokenizer.from_pretrained(model.hf_config.model_name, use_fast=False)
+
+        gptq_config = GPTQConfig(
+            bits=config["bits"],
+            dataset=config["dataset"],
+            group_size=config["group_size"],
+            damp_percent=config["damp_percent"],
+            static_groups=config["static_groups"],
+            true_sequential=config["true_sequential"],
+            desc_act=config["desc_act"],
+            sym=config["sym"],
+            tokenizer=tokenizer,
+        )
+
+        def get_onnx_quant_linear(*args, **kwargs):
+            return QuantLinearORT
+
+        import auto_gptq
+
+        # GPTQ Quantization in transformers use auto_gptq under the hood, so we
+        # replace QuantLinear in auto_gptq with QuantLinearORT for quant linear layer packing
+        auto_gptq.utils.import_utils.dynamically_import_QuantLinear = get_onnx_quant_linear
+        quantized_model = AutoModelForCausalLM.from_pretrained(
+            model.hf_config.model_name,
+            quantization_config=gptq_config,
+            attn_implementation="eager",
+            torch_dtype=torch.half,
+        )
+
+        onnx_export_from_model(
```

Comment:
I don't think we should keep integrating optimum package func into Olive. Any specific reason we need to use this func?

Reply:
I tried conversion pass in olive followed by quantized pytorch model, but can't run text generation task like humaneval with expected accurary. Optimum export is used here to keep accuray combined with model.generate api.

Reply:
I don't want to use our conversion pass here. What about the standard torch-onnx export api?

Reply:
Didn't try, maybe not easy to do with torch onnx. As far as I know, it needs to export two graphs and merge with an if node to do prompt processing and token generation.

Reply:
Yeah, it would not be straightforward to use torch.export since we need io_config and dummy input. 
One way could be to break this workflow into quantization (save torch model) and optimum conversion pass. but the torch model probably cannot be saved to disk because of the module replacement done here? 

If so, I think it's okay to use optimum export for now.

Reply:
@xiaoyu-work, https://github.com/microsoft/Olive/blob/main/olive/passes/onnx/optimum_conversion.py#L69 use the similar manner to import Optimum functions.

@Taka152 , please add version check since the onnx_export_from_model only exist after Optimum 1.17

Reply:
Yeah I believe one of our long term goals is to decoupling Olive from optimum. So eventually optimum conversion will be removed from Olive also. I'm okay with this for now but we should try to avoid using optimum package.

Reply:
@guotuofeng Can't use main_export because it don't support model with replaced module. It need to add onnx_export from_model func and parameters to optimum_conversion. I'll add a version check.

@jambayk I tried to breaking into two passes, it will bring changes in user_script.py to use dummpy inputs and dataloader without hf config and some other hackings to run evaluation for module replace pytorch model. Finally for a clean PR, I choose to use one pass.

Reply:
I could try other generation solution like onnxruntime-genai to decouple optimum step by step after this PR.